text_generation/causal_lm/cpp/README.md

```diff
@@ -4,6 +4,26 @@ These examples showcase inference of text-generation Large Language Models (LLMs
 
 ## How it works
 
+### Stateful LLM
+
+The examples infer a stateful model. The stateful model keeps internal state which is updated after every inference. That state is used as additional implicit model inputs. It is opposed to stateless model which requires passing state tensors explicitly. A more detailed introduction to stateful models can be found in https://github.com/openvinotoolkit/openvino/pull/22410. TODO: update link
+
+The intermediate inference results are usually stored in KV (key value) cache in case of LLMs. This cache is represented by the corresponding inputs and outputs. Stateless model approach requires manipulating these outputs to use them as inputs for the next inference. Stateful models remove that burden and keep KV cache internally. Although the cache is still accessible with state API.
```

Comment:
```suggestion
The intermediate inference results are usually stored in KV (key/value) cache in case of LLMs. This cache is represented by the corresponding inputs and outputs in model implemented originally in DL framework (e.g. PyTorch models from HuggingFace). Stateless model approach requires manipulating these outputs to use them as inputs for the next inference. Stateful models remove that burden and keep KV cache internally. Although the cache is still accessible with state API.
```

Also fuse with previous paragraph to avoid an extra LLM-independent education on stateful models. Start with LLMs specific first to give the domain context, and then go to stateful terms and compare it with stateless.

Try to borrow a few sentences from https://docs.openvino.ai/2023.3/gen_ai_guide.html

Reply:
Done